### PR TITLE
[RTL] Add ShlOp.

### PR DESCRIPTION
**Pull request by cgyurgyik**
Piece of #9 .

### CODE REVIEW

File: include/circt/Dialect/RTL/Combinatorial.td [hunk 1]
```diff
@@ -94,6 +94,7 @@ def SubOp : UTBinRTLOp<"sub">;
 def MulOp : UTVariadicRTLOp<"mul", [Commutative]>;
 def DivOp : UTBinRTLOp<"div">;
 def ModOp : UTBinRTLOp<"mod">;
+def ShlOp : UTBinRTLOp<"shl">;
```
Comment (lattner) [inline include/circt/Dialect/RTL/Combinatorial.td:97]: `UTBinRTLOp` implies that the LHS and RHS of the binary operator are the same width.  We might eventually want to relax this for something like a dynamic shift.  That said, we can continue to discuss this and refine in subsequent patches.

File: lib/EmitVerilog/EmitVerilog.cpp [hunk 2]
```diff
@@ -849,6 +849,9 @@ class ExprEmitter : public ExprVisitor<ExprEmitter, SubExprInfo>,
   SubExprInfo visitComb(rtl::ModOp op) {
     return emitSignedBinary(op, Multiply, "%");
   }
+  SubExprInfo visitComb(rtl::ShlOp op) {
+    return emitSignedBinary(op, Shift, "<<");
```
Comment (lattner) [inline lib/EmitVerilog/EmitVerilog.cpp:853]: Please add a testcase for this in EmitVerilog! :-)

**Review by lattner — APPROVED** [review]
Looks really great!

File: lib/EmitVerilog/EmitVerilog.cpp (hunk 2 above)
Comment (cgyurgyik) [reply lib/EmitVerilog/EmitVerilog.cpp:853]: Done, PTAL. I wasn't really sure how the naming schema for the modules worked, so just threw a 'C' on it.

**Comment by lattner** [comment]
LGTM.  i've just been adding testcases to the large module above, but adding new modules is also great.  I'd rename it from C to something more specific like "shifts" or shl depending on what you want this to eventually contain.  Otherwise LGTM!

**Pull request merged**